debezium-core/src/main/java/io/debezium/relational/TableSchemaBuilder.java

```diff
@@ -106,13 +108,22 @@ public TableSchema create(String schemaPrefix, String envelopSchemaName, Table t
                 // Add the column to the value schema only if the column has not been filtered ...
                 ColumnMapper mapper = mappers == null ? null : mappers.mapperFor(tableId, column);
                 addField(valSchemaBuilder, column, mapper);
+                addField(proxyKeySchemaBuilder, column, mapper);
             }
         });
         Schema valSchema = valSchemaBuilder.optional().build();
         Schema keySchema = hasPrimaryKey.get() ? keySchemaBuilder.build() : null;
+        Schema proxyKeySchema = proxyKeySchemaBuilder.build();
 
+        if (keySchema == null) {
+            LOGGER.info("Table '{}' does not have primary key, mapping it to proxy key with schema: {}", tableId, SchemaUtil.asDetailedString(proxyKeySchema));
```

Comment:
I don't think "proxy" is the right name here. Rather "surrogate" or something like that.